File: server/medicore.api/src/MediCore.Api/InputParam/UserAppointmentViewModel.cs
using MediCore.Data.Infrastructure;
using MediCore.Service.Model;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Api.InputParam
{
    /// <summary>
    /// User appointment
    /// </summary>
    public class UserAppointmentViewModel
    {
        public int Id { get; set; }
        public AppointmentStatus Status { get; set; }
        [Required(ErrorMessage = "Required")]
        public DateTime AppointmentDate { get; set; }
        [Required(ErrorMessage = "Required")]
        public int HospitalId { get; set; }
        [Required(ErrorMessage = "Required")]
        public int MedicalStaffId { get; set; }

        public List<UserPatientAppointmentModel> PatientProblems { get; set; }
    };

    public class UserPatientAppointmentModel
    {
        [Required(ErrorMessage = "Required")]

        public int PatientId { get; set; }

        [Required(ErrorMessage = "Required")]
        public string Problems { get; set; }
    }
}

File: server/medicore.apps/src/model/userAppointmentViewModel.ts
import { AppointmentStatus, UserPatientAppointmentModel } from '../model';
import { IEntityBase } from './IEntityBase';

export class UserAppointmentViewModel implements IEntityBase {
    id: number;
    status: AppointmentStatus;
    appointmentDate: Date;
    hospitalId: number;
    medicalStaffId: number;
    patientProblems: UserPatientAppointmentModel[];

    constructor() {
        this.id = 0;
        this.status = null;
        this.appointmentDate = null;
        this.hospitalId = 0;
        this.medicalStaffId = 0;
        this.patientProblems = null;
    }
}

